irohad/consensus/yac/storage/impl/yac_common.cpp

```diff
@@ -25,29 +25,27 @@ namespace iroha {
   namespace consensus {
     namespace yac {
 
-      bool sameProposals(const std::vector<VoteMessage> &votes) {
+      bool sameKeys(const std::vector<VoteMessage> &votes) {
         if (votes.empty()) {
           return false;
         }
 
         auto first = votes.at(0);
         return std::all_of(
             votes.begin(), votes.end(), [&first](const auto &current) {
-              return first.hash.proposal_hash == current.hash.proposal_hash;
+              return first.hash.vote_round == current.hash.vote_round;
             });
       }
 
-      boost::optional<ProposalHash> getProposalHash(
-          const std::vector<VoteMessage> &votes) {
-        auto &&hash = getHash(votes);
-        if (hash) {
-          return (*hash).proposal_hash;
+      boost::optional<Round> getKey(const std::vector<VoteMessage> &votes) {
+        if (not sameKeys(votes)) {
```

Comment:
`.. or votes.size() == 0`

Reply:
No, because vote messages can contain different keys.

Reply:
This is checked inside `sameKeys()`